Remove PrefersNonDefaultGPU from desktop file

Jan200101 opened the pull request:
PrefersNonDefaultGPU is currently very broken, see https://github.com/ValveSoftware/steam-for-linux/issues/9940

initially added by #16

Eonfge commented:
It isn't. In fact, I'm the one that added it to Steam. Not all display managers properly support it though.

Jan200101 commented:
I'm sad to say that it is broken and I've spend almost 2 years now trying to fix it
and its not just here but also on Steam hence why its patched on flathub and some distros

https://github.com/flathub/com.valvesoftware.Steam/blob/c87b10fad237c9bf1e201d2d36fc2e80cea487b6/com.valvesoftware.Steam.yml#L185

if there is anything unclear I can give a full explanation on whatm, where, how and why but the most important bits can already be gleemed from the steam-for-linux issue linked in the initial description

Eonfge commented:
https://specifications.freedesktop.org/desktop-entry-spec/latest-single/

Still part of the specification, and I think that most people with problems use immature desktop environments or closed source graphics drivers.

Jan200101 commented:
> Still part of the specification,

correct, it was proposed in 2019 and added in 2020
https://gitlab.freedesktop.org/xdg/xdg-specs/-/merge_requests/13
and the description can be found in the recognized keys section
https://specifications.freedesktop.org/desktop-entry-spec/latest/recognized-keys.html

> PrefersNonDefaultGPU
> If true, the application prefers to be run on a more powerful discrete GPU if available, which we describe as “a GPU other than the default one” in this spec to avoid the need to define what a discrete GPU is and in which cases it might be considered more powerful than the default GPU. This key is only a hint and support might not be present depending on the implementation. 

this is where the issue is
the first part of the the description is how the key is intended to be used and how everyone is using it
> If true, the application prefers to be run on a more powerful discrete GPU if available

however its been implemented by everyone in accordance to what followed
> which we describe as “a GPU other than the default one” in this spec

even if we ignore the possible intent of the spec, the creator explicitly outlines how its suppose to function in a comment

![image](https://github.com/user-attachments/assets/4548f9f4-5f8d-4b47-83a8-b1d9a2446767)

> If reality bites, and we have systems where the more powerful GPU is the default one, I'd expect the implementation to figure that out.

No implementation has figured that out, I've proposed a fix but nothing has materialized from it.

> and I think that most people with problems use immature desktop environments or closed source graphics drivers.

The desktop key is generally implemented through swicheroo-control which is a dbus service created to query GPU information.

If you do a search for `net.hadess.SwitcherooControl` on github you'll find
- Gnome
- KDE
- Bungie
- Cinnamon
all support the option

further, I am not sure how the feature being not supported is relevant?
its implementation actively breaks setups so if its not supported this wouldn't happen.

and even those that do implement it implement it wrong!
and even those that do implement it implement it incorrectly which is why this is an issue in the first place.

closed source drivers aren't an issue here either, switcheroo-control has a special case for the proprietary Nvidia driver where it ends up using the nvidia specific environment variables instead of the Mesa ones
https://gitlab.freedesktop.org/hadess/switcheroo-control/-/blob/370009087072fbf7d30c381bac9457ea735b83c8/src/switcheroo-control.c#L265

---

I think that if the keys works as intended it would be the best for everyone however it does not and you are forced to either screw over Hybrid Laptops or Desktop users

a fix to switcheroo-control (and GNOME, KDE, Cinnamon) has been proposed almost 2 years ago that was then reworked and cleaned up a year ago to no avail due to RedHat shifting responsibilities leaving the project unmaintained.

https://gitlab.freedesktop.org/hadess/switcheroo-control/-/merge_requests/69

forking would be a long term alternative but short term it would only prolong things until I can drive adoption by distro maintainers which would convince desktop developers to adopt the changes and until then the best I can do is convince developers that the key does more harm than good

Eonfge commented:
Done with https://github.com/flathub/org.zdoom.GZDoom/pull/93
